src/Microsoft.Azure.NotificationHubs/NotificationHubClient.cs

```diff
@@ -2186,7 +2156,21 @@ private void SetUserAgent()
                 xmlReader.ReadToDescendant("content");
                 xmlReader.ReadStartElement();
 
-                return (TEntity)_entitySerializer.Deserialize(xmlReader, xmlReader.Name);
+                var entity = _entitySerializer.Deserialize(xmlReader, xmlReader.Name);
+
+                if (entity is GcmTemplateRegistrationDescription)
+                {
+                    var fcmTemplateRegistrationDescription = new FcmTemplateRegistrationDescription(entity as GcmTemplateRegistrationDescription);
+                    return (fcmTemplateRegistrationDescription as TEntity);
+                }
+
+                if (entity is GcmRegistrationDescription)
```

Comment:
Same as above `entity is GcmRegistrationDescription gcmRegistration`

Reply:
have fixed too